### PR TITLE
DO-1530: Upgrade constructs

### PR DESCRIPTION
**Pull request by gowrizrh**
**Description of the proposed changes**  

* Create new Esbuild package
* Upgrade `basic-auth` construct to CDK v2
* Upgrade `cloudfront-security-headers` to CDK v2
* Upgrade `geoip-redirect` to CDK v2
* Update dependencies of `lambda-at-edge-handlers`
* Upgrade `prerender-proxy` to CDK v2
* Upgrade `rabbitmq` to CDK v2
* Upgrade `shared-vpc` to CDK v2
* Upgrade `waf` to CDK v2

### CODE REVIEW

**Review by TheOrangePuff — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
   },
   "homepage": "https://github.com/aligent/aws-cdk-constructs#readme",
   "scripts": {
-    "build": "",
-    "prepublish": "",
+    "build": "tsc",
+    "prepublish": "tsc",
```
[the remark this answers is not in the document]
Comment (gowrizrh) [reply package.json:22]: this currently exists in the main branch. i only removed the commands for convenience and then re-added them back :thinking:

File: packages/cloudfront-security-headers/test/cdk-cloudfront-security-headers.test.ts
[the remark this answers is not in the document]
Comment (gowrizrh) [reply packages/cloudfront-security-headers/test/cdk-cloudfront-security-headers.test.ts:1]: probably not, i'm not sure if this was even right, but i didn't spend anytime migrating tests :stuck_out_tongue:

File: packages/cloudfront-security-headers/lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  readonly edgeFunction: EdgeFunction;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    props?: SecurityHeaderFunctionProps
+  ) {
+    super(scope, id);
+
+    const defineOptions: any = {};
```
[the remark this answers is not in the document]
Comment (gowrizrh) [reply packages/cloudfront-security-headers/lib/index.ts:22]: i think we could add `__CONTENT_SECURITY_POLICY__` :thinking:

**Comment by gowrizrh** [comment]
thanks @TheOrangePuff, i'll make the suggested changes in a separate PR

**Pull request merged**